lightningd/lightningd.h

```diff
@@ -70,6 +70,12 @@ struct config {
 
 	/* EXPERIMENTAL: offers support */
 	bool exp_offers;
+
+	/* Allow dust reserves (including 0) when being called via
+	 * `fundchannel` or in the `openchannel` hook. This is a
+	 * slight spec incompatibility, but implementations do this
+	 * already. */
+	bool allowdustreserve;
```

Comment:
I'm ok with this, but....

We should check we never add a (dust) HTLC which turns all outputs to dust (zero-output txs are invalid).  We should neither propose such a thing (using peer's dust limit), nor accept it.

One simple way is to simply decline to allow opens where the channel size is less than (my-max_accepted_htlcs + their-max_accepted_htlcs + 2) * min(my-dust_limit_satoshis, their-dust_limit_satoshis)?

Reply:
No. Without a reserve, *all* our outputs could be dust, which is illegal.